environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/instance/enhanced/selectType/WeldInstanceSelectTypeTest.java

```diff
@@ -0,0 +1,84 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.environment.se.test.instance.enhanced.selectType;
+
+import java.lang.reflect.Type;
+
+import org.jboss.arquillian.container.se.api.ClassPath;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.BeanArchive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.weld.environment.se.Weld;
+import org.jboss.weld.environment.se.WeldContainer;
+import org.jboss.weld.inject.WeldInstance;
+import org.jboss.weld.test.util.Utils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * WeldContainer already gives access to WeldInstance - this tests that you can select based on Type
+ *
+ * @author <a href="mailto:manovotn@redhat.com">Matej Novotny</a>
+ */
+@RunWith(Arquillian.class)
+public class WeldInstanceSelectTypeTest {
+
+    @Deployment
+    public static Archive<?> createTestArchive() {
+        return ClassPath.builder().add(ShrinkWrap.create(BeanArchive.class, Utils.getDeploymentNameAsHash(WeldInstanceSelectTypeTest.class))
+            .addPackage(WeldInstanceSelectTypeTest.class.getPackage())).build();
+    }
+
+    @Test
+    public void testValidSelect() {
+        try (WeldContainer container = new Weld().initialize()) {
+            // Class<T> implements Type, so let's make use of that
+            Type firstType = SomeInterface.class;
+
+            // Instance<Object> -> Instance<SomeInterface>
+            WeldInstance<Object> instanceObject = container.select(firstType);
+            Object object = instanceObject.get();
+
+            // this can be casted if needed
+            Assert.assertTrue(object instanceof SomeInterface);
+            SomeInterface si = (SomeInterface) object;
+            Assert.assertEquals(SomeOtherBean.class.getSimpleName(), si.ping());
+
+        }
+    }
+
+    @Test
+    public void testInvalidSelect() {
+        Type secondType = SomeOtherBean.class;
+        try (WeldContainer container = new Weld().initialize()) {
+            // get more specific instance than Object and try again
+            WeldInstance<SomeInterface> someInterfaceInstance = container.select(SomeInterface.class);
+            // selecting from anything else than Object is not allowed
+            try {
+                someInterfaceInstance.select(secondType);
+                Assert.fail("Selecting by type from any other type than Object should fail with IllegalStateException!");
+            } catch (IllegalStateException e) {
+                // expected
+            }
+        }
+    }
+    
+    // TODO add tests which will use a @Typed bean eliminating Object as bean type
```

Comment:
I would also add a test that `WeldInstance.destroy()` works if you create a new `@Dependent` bean instance with the new `select()` method.